src/main/java/org/cirdles/topsoil/app/utils/GetDocumentsDirectoryOperation.java

```diff
@@ -27,29 +27,60 @@
 
     @Override
     protected Path performOnWindows(String... params) {
-        if (params.length == 1 && !"".equals(params[0])) {
-            return Paths.get(System.getProperty("user.home"), "Documents", params[0]);
-        } else {
-            return Paths.get(System.getProperty("user.home"), "Documents");
-        }
+
+        return getPath(params);
     }
 
     @Override
     protected Path performOnMacOS(String... params) {
-        if (params.length == 1 && !"".equals(params[0])) {
-            return Paths.get(System.getProperty("user.home"), "Documents", params[0]);
-        } else {
-            return Paths.get(System.getProperty("user.home"), "Documents");
-        }
+
+        return getPath(params);
     }
 
     @Override
     protected Path performOnLinux(String... params) {
-        if (params.length == 1 && !"".equals(params[0])) {
-            return Paths.get(System.getProperty("user.home"), "Documents", params[0]);
-        } else {
+
+        return getPath(params);
+    }
+
+    /**
+     * 
+     * @param params  The folders within the users documents.
+     * @returns A path object representing the directory.
+     */
+
+    private Path getPath(String... params){
+
+      String param = validateParams(params);
+
+      if (params.length == 1 && !"".equals(param)) {
+            return Paths.get(System.getProperty("user.home"), "Documents", param);
+        }
+      else {
             return Paths.get(System.getProperty("user.home"), "Documents");
         }
     }
```

Comment:
This method seems a little busy. Maybe add a method `Path performOnAll(String[] params)` that unwraps the appropriate values and delegates to `Path getPath()` and `Path getPath(String directoryName)`.